packages/typescript-estree/src/parseSettings.ts

```diff
@@ -0,0 +1,317 @@
+import debug from 'debug';
+import { sync as globSync } from 'globby';
+import isGlob from 'is-glob';
+import type * as ts from 'typescript';
+
+import type { CanonicalPath } from './create-program/shared';
+import {
+  ensureAbsolutePath,
+  getCanonicalFileName,
+} from './create-program/shared';
+import { inferSingleRun } from './inferSingleRun';
+import { log } from './log';
+import type { TSESTreeOptions } from './parser-options';
+import type { TSESTree } from './ts-estree';
+import { warnAboutTSVersion } from './warnAboutTSVersion';
+
+type DebugModule = 'typescript-eslint' | 'eslint' | 'typescript';
+
+/**
+ * Internal settings used by the parser to run on a file.
+ */
+export interface ParseSettings {
```

Comment:
one thing i'd love is for us to enforce that this object is immutable now to ensure we're not mutating it anywhere outside this file.